### PR TITLE
Commands that start with s

### PR DESCRIPTION
**Pull request by Exr0n**
Fixed `SafteyCommand.java`, `SingleOp.java`, `SpawnCommand.java` to use `CommandBase`.

### CODE REVIEW

File: commands/SafetyCommand.java [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
 	public SafetyCommand(double timeout) {
-		super(timeout);
+		super();
+		withTimeout(timeout);
 	}
 
 	public SafetyCommand(String name, double timeout) {
-		super(name, timeout);
+		super();
+		setName(name);
```
Comment (NikhilSuresh24) [inline commands/SafetyCommand.java]: don't implement a timeout method, i changed my mind

File: commands/SafetyCommand.java [hunk 2]
```diff
@@ -1,35 +1,38 @@
...
+ * A command that will only run if the isSafe() method returns safe. To use,
+ * implement an isSafe() method, and use executeIfSafe() in place of the
+ * execute() method
  */
-public abstract class SafetyCommand extends Command {
+public abstract class SafetyCommand extends CommandBase {
 	protected String reasonUnsafe;
 
 	public SafetyCommand() {
 		super();
```
Comment (NikhilSuresh24) [inline commands/SafetyCommand.java]: `this("Safety Command")`

File: commands/SingleOp.java [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
+public class SingleOp extends CommandBase {
 	protected Runnable op;
 
+	public SingleOp(String name, Runnable op) {
+		setName(name);
+		this.op = op;
+	}
+
 	public SingleOp(Runnable op) {
 		this.op = op;
```
Comment (NikhilSuresh24) [inline commands/SingleOp.java]: `this("SingleOp);`

File: commands/SpawnCommand.java [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
-	
-	@Override
-	protected void interrupted() {}
+
+	protected void end() {
+	}
+
+	protected void interrupted() {
+	}
 }
```
Comment (NikhilSuresh24) [inline commands/SpawnCommand.java]: get rid of these

File: commands/SpawnCommand.java [hunk 5]
```diff
@@ -1,32 +1,34 @@
...
+		childCommand.schedule();
 	}
-	
+
 	@Override
-	protected void execute() {}
-	
+	public void execute() {
+	}
+
```
Comment (NikhilSuresh24) [inline commands/SpawnCommand.java]: no need to implement executre

File: commands/SafetyCommand.java (hunk 1 above)
Comment (Exr0n) [reply commands/SafetyCommand.java]: Done, please check again :)

File: commands/SafetyCommand.java [hunk 6]
```diff
@@ -1,35 +1,27 @@
...
-	public SafetyCommand(double timeout) {
-		super(timeout);
+		super();
+		setName(name);
 	}
 
-	public SafetyCommand(String name, double timeout) {
-		super(name, timeout);
+	public SafetyCommand() {
+		this("SafteyCommand");
```
Comment (NikhilSuresh24) [inline commands/SafetyCommand.java]: spelled safety wrong smh :(

File: commands/SafetyCommand.java [hunk 7]
```diff
@@ -1,35 +1,27 @@
...
 	}
 
 	@Override
-	protected final void execute() {
+	final public void execute() {
```
Comment (NikhilSuresh24) [inline commands/SafetyCommand.java]: public final

File: commands/SpawnCommand.java [hunk 8]
```diff
@@ -1,32 +1,24 @@
 package org.usfirst.frc4904.standard.commands;
 
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.Command;
 
-import edu.wpi.first.wpilibj.command.Command;
-
-public class SpawnCommand extends Command {
+public class SpawnCommand extends CommandBase {
 	private final Command childCommand;
```
Comment (bcho04) [inline commands/SpawnCommand.java]: `childCommand` should be of type `CommandBase`

File: commands/SafetyCommand.java [hunk 9]
```diff
@@ -1,35 +1,27 @@
...
-	public SafetyCommand(double timeout) {
-		super(timeout);
+		super();
+		setName(name);
 	}
 
-	public SafetyCommand(String name, double timeout) {
-		super(name, timeout);
+	public SafetyCommand() {
+		this("SafetyCommand");
```
Comment (Daniel-E-B) [inline commands/SafetyCommand.java]: what happened to the timeout?

File: commands/SingleOp.java [hunk 10]
```diff
@@ -1,22 +1,26 @@
...
 
-import edu.wpi.first.wpilibj.command.Command;
-
-public class SingleOp extends Command {
+public class SingleOp extends CommandBase {
 	protected Runnable op;
 
-	public SingleOp(Runnable op) {
+	public SingleOp(String name, Runnable op) {
+		setName(name);
```
Comment (Daniel-E-B) [inline commands/SingleOp.java]: do you want to call `super()` here?

Comment (Exr0n) [reply commands/SingleOp.java]: I guess? I think somebody told me to remove `super()` calls earlier.

File: commands/SafetyCommand.java (hunk 9 above)
Comment (Exr0n) [reply commands/SafetyCommand.java]: Fixed.

**Pull request merged**